src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -307,8 +330,13 @@ public RealJenkinsRule withTimeout(int timeout) {
      * might accidentally be shared across otherwise distinct services.
      * <p>Calling this method does <em>not</em> change the fact that Jenkins will be configured to listen only on localhost for security reasons
      * (so others in the same network cannot access your system under test, especially if it lacks authentication).
+     * <p>
+     * When using HTTPS, use @{link #https(String)} instead.
      */
     public RealJenkinsRule withHost(String host) {
+        if (https) {
+            throw new IllegalStateException("Don't call this method when using HTTPS");
+        }
```

Comment:
so why go to all the lengths to generate an SSL cert if it is always going to be localhost?